src/nvim/lua/converter.c

```diff
@@ -388,12 +388,12 @@ bool nlua_pop_typval(lua_State *lstate, typval_T *ret_tv)
       LuaCFunctionState *state = xmalloc(sizeof(LuaCFunctionState));
       state->lua_callable.func_ref = nlua_ref_global(lstate, -1);
 
-      char_u *name = register_cfunc(&nlua_CFunction_func_call,
-                                    &nlua_CFunction_func_free,
-                                    state);
+      char *name = (char *)register_cfunc(&nlua_CFunction_func_call,
+                                          &nlua_CFunction_func_free,
+                                          state);
 
       cur.tv->v_type = VAR_FUNC;
-      cur.tv->vval.v_string = (char *)vim_strsave(name);
+      cur.tv->vval.v_string = name;
```

Comment:
I believe this duplication is necessary, otherwise the entry in the function hash table will be freed.